File: CsLibrary/temp/Byte.cs
using System;

namespace JavaClasses {

    /**
     * The Byte class wraps a value of primitive type byte in an object.
     *
     * @author Petr Kozler (A13B0359P)
     */
    public class Byte {

        /**
         * A constant holding the maximum value a byte can have, 2^7-1.
         */
        public const sbyte MAX_VALUE = 127;

        /**
         * A constant holding the minimum value a byte can have, -2^7.
         */
        public const sbyte MIN_VALUE = -128;

        private sbyte v;

        /**
         * Constructs a newly allocated Byte object that represents the specified byte value.
         */
        public Byte(sbyte value) {
            v = value;
        }

        /**
         * Returns the value of this Byte as a byte.
         */
        public sbyte byteValue() {
            return v;
        }

        /**
         * Compares two Byte objects numerically.
         */
        public int compareTo(Byte anotherByte) {
            return compare(v, anotherByte.v);
        }

        /**
         * Compares two byte values numerically.
         */
        public static int compare(sbyte x, sbyte y) {
            /* positive number if x is greater, 0 if x equals y, negative number otherwise
               (same as for char and short types) */
            return x - y;
        }

        /**
         * Compares this object to the specified object.
         */
        public bool equals(Object obj) {
            if (obj == null) {
                return false;
            }

            if (GetType() != obj.GetType()) {
                return false;
            }

            return v.Equals((obj as Byte).v);
        }

        public override string ToString() {
            return toString().ToString();
        }

        /**
         * Returns a String object representing this Byte's value.
         */
        public String toString() {
            return Byte.toString(v);
        }

        /**
         * Returns a new String object representing the specified byte.
         */
        public static String toString(sbyte b) {
            return b.ToString();
        }

        /**
         * Parses the string argument as a signed decimal byte.
         */
        public static sbyte parseByte(String s) {
            return sbyte.Parse(s.ToString());
        }
    }
}

File: CPlusPlus/CppLibrary/Byte.cpp
#include "Byte.h"

#include <string>
#include <cstdlib>

const int8_t Byte::MAX_VALUE = 127;
const int8_t Byte::MIN_VALUE = -128;

/**
 * Constructs a newly allocated Byte object that represents the specified
 * byte value.
 *
 * @param value the value to be represented by the Byte.
 */
Byte::Byte(int8_t value) {
    v = value;
}

/**
 * Returns the value of this Byte as a byte.
 *
 * @return the numeric value represented by this object after conversion to
 * type byte.
 */
int8_t Byte::byteValue() {
    return v;
}

/**
 * Compares two Byte objects numerically.
 *
 * @param anotherByte the Byte to be compared.
 * @return the value 0 if this Byte is equal to the argument Byte; a value
 * less than 0 if this Byte is numerically less than the argument Byte; and
 * a value greater than 0 if this Byte is numerically greater than the
 * argument Byte (signed comparison).
 */
int32_t Byte::compareTo(Object *anotherByte) {
    return compare(v, ((Byte *)anotherByte)->v);
}

/**
 * Compares two byte values numerically.
 *
 * @param x the first byte to compare
 * @param y the second byte to compare
 * @return the value 0 if x == y; a value less than 0 if x < y; and a value greater than 0 if x
 * > y
 */
int32_t Byte::compare(int8_t x, int8_t y) {
    // positive number if x is greater, 0 if x equals y, negative number otherwise
    return x - y;
}

/**
 * Compares this object to the specified object.
 *
 * @param obj the object to compare with
 * @return true if the objects are the same; false otherwise.
 */
bool Byte::equals(Object *obj) {
    // testing another object reference for a NULL value
    if (obj == nullptr) {
        return false;
    }

    // testing object class equality
    if (sizeof(this) != sizeof(*obj)) {
        return false;
    }

    // testing object fields equality
    return (v == ((Byte *)obj)->v);
}

/**
 * Returns a String object representing this Byte's value.
 * 
 * @return a string representation of the value of this object in base 10.
 */
String *Byte::toString() {
    return toString(v);
}

/**
 * Returns a new String object representing the specified byte.
 *
 * @param b the byte to be converted
 * @return the string representation of the specified byte
 */
String *Byte::toString(int8_t b) {
    return new String(std::to_string(b));
}

/**
 * Parses the string argument as a signed decimal byte.
 *
 * @param s a String containing the byte representation to be parsed
 * @return the byte value represented by the argument in decimal
 */
int8_t Byte::parseByte(String *s) {
    String *str = s->trim();
    int8_t value = strtol(str->_s().c_str(), nullptr, 10);
    delete str;
    
    return value;
}
